### ISSUE TITLE
Match.js to support checking for symbol 

### ISSUE THREAD

**Issue by khamiltonuk** [sinonjs/samsam#112]
The match function is missing a case for the primitive symbol

**What did you expect to happen?**
A user can check if two values are both symbols, the match() should return true 

**What actually happens**
```js
it("returns true if two values are both symbols", () => {
        var symbol1 = Symbol("hello");
        var symbol2 = Symbol("hello2");

        var checkMatch = samsam.match(symbol1, symbol2);
        assert.isTrue(checkMatch);
});
```
Currently get the error message `Error: Matcher was not a string, a number, a function, a boolean or an object`

**Comment by mgred** [comment]
@khamiltonuk 
> **What did you expect to happen?**
> A user can check if two values are both symbols, the match() should return true

I think this assumption is not quite correct. Since these are two completly different `Symbol`s, this schould return `false`. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) even two `Symbol`s with the same value are not the same (`Symbol("foo") === Symbol("foo") // false`).

**Comment by mgred** [comment]
@khamiltonuk I created a PR (#127). Let's discuss how we proceed here 😄

**Comment by mroderick** [comment]
This has been fixed with #127
